server/claim/claim.service.ts

```diff
@@ -154,24 +154,34 @@ export class ClaimService {
         return this._getClaim({_id: claimId})
     }
 
-    async getByPersonalityIdAndClaimSlug(personalityId, claimSlug) {
-        return this._getClaim({personality: personalityId, slug: claimSlug})
+    async getByPersonalityIdAndClaimSlug(personalityId, claimSlug, revisionId = undefined) {
+        return this._getClaim({personality: personalityId, slug: claimSlug}, revisionId)
     }
 
-    private async _getClaim(match: ClaimMatchParameters, postprocess = true) {
+    private async _getClaim(match: ClaimMatchParameters, revisionId = undefined, postprocess = true) {
```

Comment:
The post-process flag is not a good idea, we need to manipulate the claim object properly so it will have the same structure when `this.postProcess` is called

Reply:
My suggestion is something like:
```
        let claim
        if(revisionId) {
            claim = {
                ...await this.ClaimModel.aggregate([
                    { $match: match },
                    { $project: { "latestRevision": 0}}
                ])[0],
                revision: await this.claimRevisionService.getRevision(revisionId)
            }
        } else {
           claim = await this.ClaimModel.findOne(match)
                .populate("personality", "_id name")
                .populate("sources", "_id link classification")
                .populate("latestRevision").toObject()
        }

        if (!claim) {
              throw new NotFoundException()
        }
        return this.postProcess(claim)
```

Reply:
if we remove the conditional  return postprocess === true ? this.postProcess(claim.toObject()) : claim;

we'll affect the update function, which doesn't need to pass in postProcess function

Reply:
and the declaration:
...await this.ClaimModel.aggregate([
                    { $match: match },
                    { $project: { "latestRevision": 0}}
                ])[0],
doesn't returns the claim

Reply:
![image](https://user-images.githubusercontent.com/73478823/163053616-0062eb1f-7e28-49dc-a62d-fbe13e7d82ea.png)
 i had to make this

Reply:
its having a lot of conflicts let's make a call